heudiconv/dicoms.py

```diff
@@ -202,7 +204,8 @@ def group_dicoms_into_seqinfos(
     flatten: Literal[False, True] = False,
     custom_grouping: str
     | Callable[
-        [list[str], Optional[Callable[[dcm.dataset.Dataset], Any]], type[SeqInfo]], Any
+        [list[str], Optional[Callable[[dcm.dataset.Dataset], Any]], type[SeqInfo]],
+        dict[SeqInfo, list[str]],
```

Comment:
grr... elderly me fails to find where I have added some type defined for another callback in heudiconv (may be I never finished/submitted that PR) following the https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols  . Because this signature is quite long - I wonder if we should just define `CustomGrouppingT` or alike to be used in all instances here, WDYT?

Reply:
Unless we're going to use this type outside of `group_dicoms_into_seqinfos()`, I don't see the need for an alias.